veniq/baselines/semi/cluster_statements.py

```diff
@@ -0,0 +1,94 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional
+
+from veniq.ast_framework import AST, ASTNode
+from veniq.baselines.semi.extract_semantic import StatementSemantic, extract_method_statements_semantic
+from veniq.baselines.semi.common_cli import common_cli
+
+Cluster = List[ASTNode]
+
+
+@dataclass
+class Range:
+    first: int
+    last: int
+
+
+def cluster_statements(statements_semantic: Dict[ASTNode, StatementSemantic]) -> List[Cluster]:
+    clusters: List[Cluster] = []
+    for step in range(1, len(statements_semantic) + 1):
+        for cluster in _StatementsClusterIterator(statements_semantic, step):
+            if cluster not in clusters:
```

Comment:
How does it decide that cluster object not in the list? How does list compare in python?

Reply:
I use linear search. In sense of speed, it can be faster than binary on small data. I decided to use it to just keep the order of clusters appearance for easier testing. I could be done with set.